hardware/DomoticzTCP.cpp

```diff
@@ -136,34 +88,32 @@ bool DomoticzTCP::StopHardware()
 
 bool DomoticzTCP::ConnectInternal()
 {
-	m_socket = socket(info->ai_family, info->ai_socktype, info->ai_protocol);
-	if (m_socket == INVALID_SOCKET)
+	if (m_connection)
+		delete m_connection;
+	m_connection = new csocket();
+	if (m_connection->connect(m_szIPAddress.c_str(), m_usIPPort))
 	{
-		_log.Log(LOG_ERROR, "Domoticz: TCP could not create a TCP/IP socket!");
+		_log.Log(LOG_ERROR, "Domoticz: TCP could not connect to: %s:%d", m_szIPAddress.c_str(), m_usIPPort);
 		return false;
 	}
-	/*
-		//Set socket timeout to 2 minutes
-	#if !defined WIN32
-		struct timeval tv;
-		tv.tv_sec = 120;
-		setsockopt(m_socket, SOL_SOCKET, SO_RCVTIMEO,(struct timeval *)&tv,sizeof(struct timeval));
-	#else
-		unsigned long nTimeout = 120*1000;
-		setsockopt(m_socket, SOL_SOCKET, SO_RCVTIMEO, (const char*)&nTimeout, sizeof(DWORD));
-	#endif
-	*/
-	// connect to the server
-	int nRet;
-	nRet = connect(m_socket, info->ai_addr, info->ai_addrlen);
-	if (nRet == SOCKET_ERROR)
+#ifndef WIN32
```

Comment:
Can you make a 'SetKeepAlive' function in csocket ? I think other implementations could benefit from this in the feature